tls/extensions/s2n_client_key_share.c

```diff
@@ -342,21 +339,34 @@ static int s2n_client_key_share_recv_pq_hybrid(struct s2n_connection *conn, stru
         return S2N_SUCCESS;
     }
 
+    bool is_hybrid_share_length_prefixed = 0;
+    uint16_t actual_hybrid_share_size = key_share->blob.size;
+
+    /* The length of the hybrid key share must be one of two possible lengths. Its internal values are either length
+     * prefixed, or they are not. If actual_hybrid_share_size is not one of these two lengths, then
+     * s2n_is_tls13_hybrid_kem_length_prefixed() will return an error. */
+    s2n_result result = s2n_is_tls13_hybrid_kem_length_prefixed(actual_hybrid_share_size, kem_group, &is_hybrid_share_length_prefixed);
+
     /* Ignore KEM groups with unexpected overall total share sizes */
-    if (key_share->blob.size != kem_group->client_share_size) {
+    if (s2n_result_is_error(result)) {
```

Comment:
Is s2n_is_tls13_hybrid_kem_length_prefixed called anywhere else? Why make the check if we're only going to ignore the result?

Reply:
Can't we just assume no prefixing if the actual size exactly matches the unprefixed size, and otherwise carry on like before this change?

Reply:
The function used to be called in multiple places, but after refactorings during this PR, it's only called in a single place. Merged it back into the caller's function.